packages/web3-providers-ws/test/unit/constructor.test.ts

```diff
@@ -1,10 +1,30 @@
 import Web3ProviderWS from '../../src/index';
-import { WebSocketOptions } from '../../src/types';
 
 describe('Web3ProviderWS Tests', () => {
-    it('should error because invalid WS address', () => {
+    let provider: Web3ProviderWS;
+    it('should throw error for invalid WS address', () => {
         expect(() => {
             new Web3ProviderWS('http://127.0.0.1:8545');
         }).toThrowError('Invalid WebSocket URL provided');
     });
+
+    test('should instantiate with valid client URL', async (done) => {
+        let client = 'ws://127.0.0.1:8545';
+        provider = new Web3ProviderWS(client);
+        expect(provider.web3Client).toBe(client);
+        done();
+    });
+
+    test('should create instance of Web3ProviderWS and call connect', async (done) => {
+        const connectFunc = jest.spyOn(
+            Web3ProviderWS.prototype as any,
```

Comment:
Please avoid using `any`, if its the correct object and really have `connect` method it should work normal.

Reply:
made change, you are right